pkg/deploy/deploy.go

```diff
@@ -69,7 +70,12 @@ func New(ctx context.Context, log *logrus.Entry, config *RPConfig, version strin
 		return nil, err
 	}
 
-	kvAuthorizer, err := auth.NewAuthorizerFromEnvironmentWithResource(azure.PublicCloud.ResourceIdentifiers.KeyVault)
+	environment, err := azureenvironment.AzureEnvFromEnvVar()
```

Comment:
I think this should also use env.Core - is that possible?

Reply:
I think so - I thought it might be more than was needed, but it would simplify this

Reply:
done